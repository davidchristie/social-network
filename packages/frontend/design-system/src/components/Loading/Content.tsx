import CircularProgress from "@material-ui/core/CircularProgress";
import { WithStyles } from "@material-ui/core/styles/withStyles";
import React from "react";

type Props = WithStyles<"loading"> & WithStyles<"progress">;

const Content: React.StatelessComponent<Props> = ({ classes }) => {
  return (
    <div className={classes.loading}>
      <CircularProgress className={classes.progress} />
    </div>
  );
};

export default Content;
