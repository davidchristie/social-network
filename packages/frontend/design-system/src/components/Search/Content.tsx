import MenuItem from "@material-ui/core/MenuItem";
import Paper from "@material-ui/core/Paper";
import { Breakpoint } from "@material-ui/core/styles/createBreakpoints";
import { WithStyles } from "@material-ui/core/styles/withStyles";
import { isWidthUp } from "@material-ui/core/withWidth/withWidth";
import match from "autosuggest-highlight/match";
import parse from "autosuggest-highlight/parse";
import React from "react";
import Autosuggest from "react-autosuggest";

import { Props, Suggestion } from ".";
import Icon from "../Icon";

type ContentProps = Props
  & WithStyles<"input">
  & WithStyles<"root">
  & WithStyles<"search">
  & WithStyles<"suggestion">
  & WithStyles<"suggestionsContainerOpen">
  & WithStyles<"suggestionsList">
  & { width: Breakpoint; };

interface ContentState {
  suggestions: Suggestion[];
  value: string;
}

function renderSuggestion (suggestion: Suggestion, { query, isHighlighted }) {
  const matches = match(suggestion.name, query);
  const parts = parse(suggestion.name, matches);
  return (
    <MenuItem component="div" selected={isHighlighted}>
      {parts.map((part, index) => {
        const style: React.CSSProperties = {
          whiteSpace: "pre-wrap",
        };
        const highlightedStyle: React.CSSProperties = {
          ...style,
          // fontWeight: "bold",
          textShadow: "0px 0px 1px rgba(0,0,0,0.5)",
        };
        return (
          <span key={index} style={part.highlight ? highlightedStyle : style}>
            {part.text}
          </span>
        );
      })}
    </MenuItem>
  );
}

function renderSuggestionsContainer (options) {
  const { containerProps, children } = options;
  return (
    <Paper {...containerProps} square={true}>
      {children}
    </Paper>
  );
}

function getSuggestionValue (suggestion) {
  return suggestion.name;
}

export default class Content extends React.Component<ContentProps, ContentState> {
  public state = {
    suggestions: [],
    value: "",
  };

  public render () {
    const { classes, width } = this.props;
    return (
      <div className={classes.root} style={{ display: isWidthUp("sm", width) ? "block" : "none" }}>
        <div className={classes.search}>
          <Icon>search</Icon>
        </div>
        {this.renderAutosuggest()}
      </div>
    );
  }

  private handleSuggestionSelected = (event, { suggestion }) => {
    this.props.onSelect(suggestion);
  }

  private handleSuggestionsFetchRequested = async ({ value }) => {
    this.setState({
      suggestions: await this.props.fetchSuggestions(value),
    });
  }

  private handleSuggestionsClearRequested = () => {
    this.setState({
      suggestions: [],
      value: "",
    });
  }

  private handleChange = async (event) => {
    const value = event.target.value;
    this.setState({
      value,
    });
  }

  private renderAutosuggest = () => {
    const { classes } = this.props;
    return (
      <Autosuggest
        getSuggestionValue={getSuggestionValue}
        inputProps={{
          className: classes.input,
          onChange: this.handleChange,
          placeholder: "Search",
          value: this.state.value,
        }}
        onSuggestionsClearRequested={this.handleSuggestionsClearRequested}
        onSuggestionsFetchRequested={this.handleSuggestionsFetchRequested}
        onSuggestionSelected={this.handleSuggestionSelected}
        renderSuggestion={renderSuggestion}
        renderSuggestionsContainer={renderSuggestionsContainer}
        suggestions={this.state.suggestions}
        theme={{
          suggestion: classes.suggestion,
          suggestionsContainerOpen: classes.suggestionsContainerOpen,
          suggestionsList: classes.suggestionsList,
        }}
      />
    );
  }
}
