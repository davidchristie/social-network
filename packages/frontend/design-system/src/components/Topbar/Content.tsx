import AppBar from "@material-ui/core/AppBar";
import { WithStyles } from "@material-ui/core/styles/withStyles";
import Toolbar from "@material-ui/core/Toolbar";
import classNames from "classnames";
import React from "react";

type Props = {
  className?: string;
} & WithStyles<"topbar">;

const Content: React.StatelessComponent<Props> = ({
  children,
  classes,
  className,
}) => {
  return (
    <div className={classNames(classes.topbar, className)}>
      <AppBar color="default" position="fixed">
        <Toolbar>
          {children}
        </Toolbar>
      </AppBar>
    </div>
  );
};

export default Content;
