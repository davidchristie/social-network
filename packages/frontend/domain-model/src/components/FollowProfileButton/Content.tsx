import { Button } from "design-system";
import React from "react";
import { MutationFn } from "react-apollo";
import {
  FollowProfile,
  FollowProfileVariables,
  UnfollowProfile,
  UnfollowProfileVariables
} from "../../generated/types";

export interface Props {
  account: {
    profile: {
      following: Array<{
        id: string;
      }>;
    };
  };
  followProfile: MutationFn<FollowProfile, FollowProfileVariables>;
  profileId: string;
  unfollowProfile: MutationFn<UnfollowProfile, UnfollowProfileVariables>;
}

const Content: React.StatelessComponent<Props> = ({
  account,
  followProfile,
  profileId,
  unfollowProfile,
}) => {
  const isFollowing = account.profile.following.some(followed => {
    return followed.id === profileId;
  });
  return isFollowing
    ? (
      <Button
        onClick={() => unfollowProfile()}
        variant="contained"
      >
        Unfollow
      </Button>
    )
    : (
      <Button
        onClick={() => followProfile()}
        style="primary"
        variant="contained"
      >
        Follow
      </Button>
    );
};

export default Content;
