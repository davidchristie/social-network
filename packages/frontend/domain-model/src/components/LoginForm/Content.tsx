import { Alert, Button, Input, Section } from "design-system";
import React from "react";
import { MutationFn } from "react-apollo";
import { Link } from "react-router-dom";
import { Login, LoginVariables } from "../../generated/types";

export interface Props {
  login: MutationFn<Login, LoginVariables>;
}

export interface State {
  email: string;
  errorMessage: string | null;
  password: string;
}

export default class Content extends React.Component<Props, State> {
  public state: State = {
    email: "",
    errorMessage: null,
    password: "",
  };

  public render () {
    return (
      <Section className="LoginForm">
        <form onSubmit={this.handleFormSubmit}>
          <h2>Login</h2>
          Don't have an account? <Link to="/signup">Click here to sign up.</Link>
          <hr />
          {this.state.errorMessage && (
            <Alert>{this.state.errorMessage}</Alert>
          )}
          <Input
            autoFocus={true}
            id="login-email"
            label="Email Address"
            name="email"
            onChange={this.handleEmailChange}
            required={true}
            type="email"
            value={this.state.email}
          />
          <Input
            id="login-password"
            label="Password"
            name="password"
            onChange={this.handlePasswordChange}
            required={true}
            type="password"
            value={this.state.password}
          />
          <Button type="submit">Submit</Button>
        </form>
      </Section>
    );
  }

  private handleEmailChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({
      email: event.target.value,
    });
  }

  private handleFormSubmit = async (event: React.FormEvent) => {
    event.preventDefault();
    const { login } = this.props;
    const { email, password } = this.state;
    try {
      await login({
        variables: {
          email,
          password,
        },
      });
    } catch (error) {
      this.setState({
        email: "",
        errorMessage: "Incorrect email or password.",
        password: "",
      });
    }
  }

  private handlePasswordChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    this.setState({
      password: event.target.value,
    });
  }
}
