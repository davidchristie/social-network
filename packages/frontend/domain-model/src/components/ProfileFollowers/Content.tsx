import { Avatar, Grid, Section } from "design-system";
import React from "react";
import { Link } from "react-router-dom";

export interface Props {
  profile: {
    followers: Array<{
      avatar: {
        url: string;
      } | null;
      id: string;
      name: string;
    }>;
  };
}

export interface State {
  anchorElement: HTMLElement | null;
}

export default class Content extends React.Component<Props, State> {
  public render () {
    const { profile } = this.props;
    return (
      <Section className="ProfileFollowers">
        <h2>Followers</h2>
        <Grid container={true} >
          {profile.followers.map(({ avatar, id, name }) => {
            return (
              <Grid item={true} key={id} sm={4} xs={12}>
                <div className="ProfileFollowers__profile">
                  <Avatar
                    image={avatar ? avatar.url : undefined}
                    size="small"
                  />
                  <Link to={`/profile/${id}`}>
                    {name}
                  </Link>
                </div>
              </Grid>
            );
          })}
        </Grid>
      </Section>
    );
  }
}
