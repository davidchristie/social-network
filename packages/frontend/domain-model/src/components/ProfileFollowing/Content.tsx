import { Avatar, Section } from "design-system";
import React from "react";
import { Link } from "react-router-dom";

export interface Props {
  profile: {
    following: Array<{
      avatar: {
        url: string;
      } | null;
      id: string;
      name: string;
    }>;
  };
}

export interface State {
  anchorElement: HTMLElement | null;
}

export default class Content extends React.Component<Props, State> {
  public render () {
    const { profile } = this.props;
    return (
      <Section className="ProfileFollowing">
        <h2>Following</h2>
        {profile.following.map(({ avatar, id, name }) => {
          return (
            <div className="ProfileFollowing__profile" key={id}>
              <Avatar
                image={avatar ? avatar.url : undefined}
                size="small"
              />
              <Link to={`/profile/${id}`}>
                {name}
              </Link>
            </div>
          );
        })}
      </Section>
    );
  }
}
