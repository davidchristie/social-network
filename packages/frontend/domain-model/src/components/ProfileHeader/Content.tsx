import { Avatar, Section } from "design-system";
import React from "react";
import FollowProfileButton from "../FollowProfileButton";

export interface Props {
  account: {
    profile: {
      id: string;
    };
  } | null;
  profile: {
    avatar: {
      url: string;
    } | null;
    id: string;
    name: string;
  };
}

const Content: React.StatelessComponent<Props> = ({ account, profile }) => {
  const isAuthenticated = Boolean(account);
  const isOwnProfile = account && account.profile.id === profile.id;
  return (
    <Section>
      <Avatar
        image={profile.avatar ? profile.avatar.url : undefined}
        size="large"
      />
      <h1>{profile.name}</h1>
      {isAuthenticated && !isOwnProfile && (
        <FollowProfileButton profileId={profile.id} />
      )}
    </Section>
  );
};

export default Content;
