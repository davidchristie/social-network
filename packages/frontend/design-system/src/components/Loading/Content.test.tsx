import React from "react";
import { itMatchesSnapshot } from "test-utilities/dist/react";
import Content from "./Content";

describe("Loading content", () => {
  const props = {
    classes: {
      loading: "loading",
      progress: "progress",
    },
  };
  itMatchesSnapshot(<Content {...props} />);
});
