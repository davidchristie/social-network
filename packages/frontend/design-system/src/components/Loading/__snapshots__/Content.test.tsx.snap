// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading content matches snapshot 1`] = `
<div
  className="loading"
>
  <WithStyles(CircularProgress)
    className="progress"
  />
</div>
`;
